oggm/utils.py

```diff
@@ -1821,6 +1822,119 @@ def compile_run_output(gdirs, path=True, monthly=False, filesuffix=''):
     return ds
 
 
+def compile_climate_input(gdirs, path=True, filename='climate_monthly',
+                          filesuffix=''):
+    """Merge the climate input files in the glacier directories into one file.
+
+    Parameters
+    ----------
+    gdirs: the list of GlacierDir to process.
+    path: where to store (default is on the working dir).
+    filename: BASENAME of the climate input files
+    filesuffix: the filesuffix of the climate file
+    """
+
+    # Get the dimensions of all this
+    rgi_ids = [gd.rgi_id for gd in gdirs]
+
+    # The first gdir might have blown up, try some others
+    i = 0
+    while True:
+        if i >= len(gdirs):
+            raise RuntimeError('Found no valid glaciers!')
+        try:
+            ppath = gdirs[i].get_filepath(filename=filename,
+                                          filesuffix=filesuffix)
+            with xr.open_dataset(ppath) as ds_clim:
+                time = ds_clim.time.values
+            break
+        except:
+            i += 1
+
+    with warnings.catch_warnings():
+        # Long time series are currently a pain pandas
+        warnings.filterwarnings("ignore", message='Unable to decode time axis')
+        ds_clim = xr.open_dataset(ppath)
+
+    if str('numpy.datetime64') in str(type(ds_clim.temp.time.values[0])):
+        y0 = ds_clim.temp.time.values[0].astype('datetime64[Y]')
+        y1 = ds_clim.temp.time.values[-1].astype('datetime64[Y]')
+    elif str('netcdftime._netcdftime.DatetimeGregorian') \
+            in str(type(ds_clim.temp.time.values[0])):
+        y0 = ds_clim.temp.time.values[0].strftime('%Y')
+        y1 = ds_clim.temp.time.values[-1].strftime('%Y')
+    time = pd.period_range('{}-10'.format(y0), '{}-9'.format(y1), freq='M')
+    pkeep = np.ones(len(time), dtype=np.bool)
+    year = time.year.values[pkeep]
+    month = time.month.values[pkeep]
+    months = month.astype(np.float32)
+    months[months == 1] = 31/365
+    months[months == 2] = 59/365
+    months[months == 3] = 90/365
+    months[months == 4] = 120/365
+    months[months == 5] = 151/365
+    months[months == 6] = 181/365
+    months[months == 7] = 212/365
+    months[months == 8] = 243/365
+    months[months == 9] = 273/365
+    months[months == 10] = 304/365
+    months[months == 11] = 334/365
+    months[months == 12] = 0
```

Comment:
This months == syntax looks weird to me. It probably all evaluates to 0/False. Did you mean to just do months[0] to months[11]?

Reply:
Yes this looks a little clumsy. I'll have a look, but I might merge this first and correct it locally in a second PR

Reply:
You could write :

m = np.cumsum([31,28,31,30,31,30,31,31,30,31,30,31])
m[-1] = 0

for i in range(12):
    months[months == (i+1)] = m[i]/365

This do exactly the same, but looks a little bit nicer.

Reply:
Thanks for your suggestions! Note that there is already some logic in utils.py about this date conversion things: https://github.com/OGGM/oggm/blob/master/oggm/utils.py#L1085-L1113

But in this case I think we can even keep the logic simpler by using pd.date_range as in process_cesm_data: https://github.com/OGGM/oggm/blob/master/oggm/core/climate.py#L205-L212 .

@anoukvlug what do you think: do you want to give it a try? Also it would be great to have a test for this [here](https://github.com/OGGM/oggm/blob/master/oggm/tests/test_prepro.py#L1848)

Reply:
Thanks for all the suggestions! I will try to improve the code and add a test for this.

Reply:
Note that it might be a bit more convoluted than I previously thought because in the end we use netCDF4 to write the data, not xarray: https://github.com/OGGM/oggm/blob/master/oggm/core/climate.py#L247

Anyways, let me know if you want to spend more time on this @anoukvlug , otherwise I can have a look

Reply:
I can spend some time on this this afternoon. 
It works to use: 
`time = date_to_year(year, month) `
to give the years in a floating format. 

@fmaussion I am not sure if I fully understand your second suggestion/ how it would makes it simpler?

Reply:
Sorry, i dont think this was a good idea. I think that the code in process_cesm_data is closer to what you need

Reply:
Sorry, i dont think this was a good idea. I think that the code in process_cesm_data is closer to what you need

Reply:
Do you mean to use than the format of "days since" and not use the same format as the files that are created with compile_run_output?

Reply:
Right, it would be better to use the same format (I m on my phone, sorry). Than the date_to_yrs function is surely useful, and also the .years and .month attributes of pd.date_range

But just do what works for you and add a test, no need to spend hours on this ;)

Reply:
(What I mean is that the test and the final result is more important than the internals)